Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -81,22 +81,26 @@ void Digitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
 {
   digits.clear();
   mDigits.clear();
-  mMCTruthContainer.clear();
+  mTrackLabels.clear();
 
   //array of MCH hits for a given simulated event
   for (auto& hit : hits) {
-    int labelIndex = mMCTruthContainer.getIndexedSize();
+    int labelIndex = mTrackLabels.getIndexedSize();
     //index for this hit
     int detID = hit.GetDetectorID();
-    processHit(hit, detID, mEventTime, labelIndex);
+    int ndigits = processHit(hit, detID, mEventTime);
+    //TODO need one label per Digit
+    // can use nDigit output of processHit
     MCCompLabel label(hit.GetTrackID(), mEventID, mSrcID);
-    mMCTruthContainer.addElementRandomAccess(labelIndex, label);
-  } //loop over hits
+    for (int i = 0; i < ndigits; ++i) {
+      mTrackLabels.addElementRandomAccess(labelIndex, label);
```

Comment:
@mwinn2 
Coming back this rather confusing business of labels...
After a discussion with Diego (who's implementing the MID counterpart), we kind of concluded that this is wrong. 
First, it's not a labelIndex, but a dataIndex or digitIndex. And it should not be inferred from the mTrackLabels array(s) but from the digit array, in our opinion.
So the association would be instead : 

for (int i=0; i<ndigits;++i) {
  int digitIndex = mDigits.size() - ndigits + i; // -ndigits as digits were already added to mDigits array during processHit, right ?
  mTrackLabels.addElementRandomAccess(digitIndex,label);
}

Does that make sense ?

Reply:
@aphecetche 
Yes, seems correct that the current implementation in the PR is wrong, a funny mixture of what was there before and what needs to be put. Let me cross check tomorrow with other code in O2 that I convince myself that it is indeed as you concluded with Diego.

Reply:
@aphecetche 
ok, putted in.